### PR TITLE
Add genomes to tree nodes

### PR DESCRIPTION
**Pull request by MatthijsRijlaarsdam**
Added the genomes to the tree structure, wrote tests and added comments. Tests are done using a tree found on wikipedia: https://en.wikipedia.org/wiki/File:NewickExample.svg. However, the code runs fast on real data as well.

### CODE REVIEW

File: backEnd/src/main/java/genome/DataContainer.java [hunk 1]
```diff
@@ -19,192 +19,193 @@
...
+
+    /**
+     * Set the data height.
+     *
+     * @param dataHeight New data height.
+     */
+    public void setDataHeight(double dataHeight) {
+        this.dataHeight = dataHeight;
+    }
+
```
Comment (Icety) [inline backEnd/src/main/java/genome/DataContainer.java]: Geen JavaDoc

**Comment by Icety** [comment]
Add the Javadoc for one method and then it looks good to me.

**Comment by jeffreyhelgers** [comment]
Checkstyle still gives 20 warnings.

File: backEnd/src/main/java/phylogenetictree/PhylogeneticNode.java [hunk 2]
```diff
@@ -24,21 +30,41 @@ public PhylogeneticNode() {
...
     /**
-     * Initialize this node from tree node.
+     * Initialize this node from tree node, recursively.
      * @param node The node.
+     * @param parent The parentNode of this node, null if root.
      * @param distance The distance.
+     * @param id The id of this node, root being 0. Incrementing breath first.
      */
-    public PhylogeneticNode(TreeNode node, double distance) {
+    public PhylogeneticNode(TreeNode node, PhylogeneticNode parent, double distance, int id) {
```
Comment (xcwfwf44) [inline backEnd/src/main/java/phylogenetictree/PhylogeneticNode.java]: Why is this changed here?

Comment (MatthijsRijlaarsdam) [inline backEnd/src/main/java/phylogenetictree/PhylogeneticNode.java]: The Parent is needed to iterate upwards through the tree in order to store the genomes at their specific treeNodes. The id is simply a way to identify nodes without a label.

File: backEnd/src/main/java/phylogenetictree/PhylogeneticNode.java [hunk 3]
```diff
@@ -54,7 +54,8 @@ public PhylogeneticNode() {
      * @param distance The distance.
      * @param id       The id of this node, root being 0. Incrementing breath first.
      */
-    public PhylogeneticNode(final TreeNode node, final PhylogeneticNode parent, final double distance, final int id) {
+    public PhylogeneticNode(final TreeNode node, final PhylogeneticNode parent,
```
Comment (xcwfwf44) [inline backEnd/src/main/java/phylogenetictree/PhylogeneticNode.java]: Why is there a parent node needed now, compared to the previous version of this constructor without that parameter.

File: backEnd/src/main/java/phylogenetictree/PhylogeneticNode.java [hunk 4]
```diff
@@ -24,25 +43,49 @@ public PhylogeneticNode() {
...
-     * @param node The node.
+     * Initialize this node from tree node, recursively.
+     *
+     * @param node     The node.
+     * @param parent   The parentNode of this node, null if root.
      * @param distance The distance.
+     * @param id       The id of this node, root being 0. Incrementing breath first.
      */
-    public PhylogeneticNode(TreeNode node, double distance) {
+    public PhylogeneticNode(final TreeNode node, final PhylogeneticNode parent,
```
Comment (xcwfwf44) [inline backEnd/src/main/java/phylogenetictree/PhylogeneticNode.java]: here

File: backEnd/src/main/java/phylogenetictree/PhylogeneticNode.java (hunk 3 above)
Comment (MatthijsRijlaarsdam) [inline backEnd/src/main/java/phylogenetictree/PhylogeneticNode.java]: see comment above

File: backEnd/src/main/java/phylogenetictree/PhylogeneticNode.java (hunk 2 above)
Comment (xcwfwf44) [inline backEnd/src/main/java/phylogenetictree/PhylogeneticNode.java]: This could probably be resolved in a cleaner way. Now this field also gets sent as data to the client during JSON translation even though its not needed.

**Comment by Nogeenaccount** [comment]
Looks good to me.

**Comment by xcwfwf44** [comment]
Ok for merging.

File: backEnd/src/main/java/phylogenetictree/PhylogeneticNode.java (hunk 2 above)
Comment (MatthijsRijlaarsdam) [inline backEnd/src/main/java/phylogenetictree/PhylogeneticNode.java]: which is why I will write a controller class, as the field is needed in the backend. It's mathematically impossible to store the genomes at the right node if I can not iterate upwards through the tree.
